src/packages/utils/src/utils/request-coordinator.ts

```diff
@@ -74,6 +85,32 @@ export class RequestCoordinator {
     }
   };
 
+  /**
+   * Stop processing tasks - calls to queue(), and resume() will reject with an error indicating
+   * that Ganache is disconnected. This is an irreversible action. If you wish to be able to resume
+   * processing, use pause() instead.
+   */
+  public stop() {
+    this.pause();
+    this.#stopped = true;
+  }
+
+  /**
+   * Finalise shutdown of the RequestCoordinator. Rejects all pending tasks in order. Should be
+   * called after all in-flight tasks have resolved in order to maintain overall FIFO order.
+   */
+  public end() {
+    for (const current of this.pending) {
+      current.reject(
+        new Error("Cannot process request, Ganache is disconnected.")
+      );
+    }
+
+    if (this.pending.length > 0) {
+      this.pending.splice(0, this.pending.length);
+    }
```

Comment:
What do you think of:

```suggestion
    while (this.pending.length) {
      this.pending.shift().reject(
        new Error("Cannot process request, Ganache is disconnected.")
      );
    }
```

Reply:
So I originally had that, but shift is more expensive, so opted for this pattern.

In saying that, I doubt it makes a material impact for any users, unless they have 100s of 1000s of tasks queued up, which probably isn't a likely case.

Reply:
Oh. Neat! Did you measure `shift()`?

Reply:
I did - my benchmark is pretty flawed, as I do setup _within_ the benchmark, but it's the same setup for both runs, so it just dilutes the difference.

```javascript
function codeUnderTest() {
    const data = [...new Array(n)].fill({});
    let item = data.shift();
    while (item !== undefined)
      item = data.shift();
}
```

vs

```javascript
function codeUnderTest() {
    const data = [...new Array(n)].fill({});
    for (let i = 0; i < data.length; i++) {
      const item = data[i];
    }
    data.length = 0;
}
```

For n = 100, this reports iterating is ~10x faster.

Reply:
Either way, I don't think that optimizing this for performance makes sense, over readability - using .shift() also aligns with the above where we are processing normally.

Assuming a case where the queue length is 1000 items (and that the total cost is the runtime of my benchmark, including allocating the data), using shift(), we would see a total of ~5ms spent iterating the array :shrug:

Reply:
Cool. Go with whatever feels more readable then!